### ISSUE TITLE
Migrate to using "pref path" for storage

### ISSUE THREAD

**Issue by sillesta** [iamgreaser/iceball#191]
Currently all game related files are written to the current directory (in this case where the executable is located). I propose we use SDL2's [`SDL_GetPrefPath()`](https://wiki.libsdl.org/SDL_GetPrefPath) instead so we can have a more permanent storage (good for screenshots etc.). This could also allow for having a lightweight launcher which downloads new versions.

This may also cause some discrepancy with how iceball is currently set up. The purpose of switching to pref path would be to separate the launcher and the executable + game data. Currently the executable and the launcher are bundled together, so that may cause some issues in how it should be structured.

Please leave feedback on how you think this should be done, if at all!

:shipit:

**Comment by Dany0** [comment]
I think this is a good idea. Here is where it points to:

*nix:
"/home/bob/.local/share/My Company/My Program Name/" 
OS X:
SDL_GetPrefPath("Srs Bznz", "Quux 2"); => /Users/fkaaman/Library/Application Support/Srs Bznz/Quux 2/
Windows
%appdata%/orgname/game

**Comment by sillesta** [comment]
Sounds good. Additionally, maybe we could find some use of [`SDL_GetBasePath()`](https://wiki.libsdl.org/SDL_GetBasePath)? Sounds more reliable than using the working directory (`SDL_GetBasePath()` returns the path where the actual executable is, not working dir)

**Comment by Dany0** [comment]
I don't think we want users to be able to move the executable around. What other reason is there to use the executable path? Put binaries in a bin folder? Why?

**Comment by rakiru** [comment]
The main one would be so that you don't have to set the executable path before launching it (such as for launching the dedi server from command-line, or a custom launcher). It's got nothing to do with users "moving the executable around".

**Comment by Dany0** [comment]
Oh right, yeah that makes sense.
